app/src/main/java/ch/deletescape/lawnchair/Utilities.java

```diff
@@ -1113,19 +1108,11 @@ public static boolean checkOutdatedLawnfeed(Context context) {
     }
 
     public static void restartLauncher(Context context) {
-        PackageManager pm = context.getPackageManager();
-        Intent startActivity = pm.getLaunchIntentForPackage(context.getPackageName());
-
-        startActivity.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
-
-        // Create a pending intent so the application is restarted after System.exit(0) was called.
-        // We use an AlarmManager to call this intent in 100ms
-        PendingIntent mPendingIntent = PendingIntent.getActivity(context, 0, startActivity, PendingIntent.FLAG_CANCEL_CURRENT);
-        AlarmManager mgr = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
-        mgr.set(AlarmManager.RTC, System.currentTimeMillis() + 100, mPendingIntent);
+        Intent startActivity = new Intent(context, Launcher.class);
+        startActivity.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+        context.startActivity(startActivity);
+        ((Activity) context).finish();
```

Comment:
Finishing and restarting an activity is not equal to restarting the launcher. There are some static references that will still persist when all activities are destroyed.

Reply:
Also, you can't just cast a Context to an Activity and expect it to always work. If you really need to, change the method signature or add a check before casting.

Reply:
ok will do after class